### DISCUSSION TITLE
Ref blocks turning into UUIDs on change of original?

### DISCUSSION THREAD

**Discussion by PaddyPatPat** [Kinneyzhang/roam-block#17] [Q&A]
Hi there,  

I really like the fundamental idea behind this. It seems like org-transclusion, but more solidly in the text.

I'm testing with test documents in my org-roam directory at the moment. I can make references between paragraphs and drop them in the other document (I can't seem to make embeds). 
- If the block-reference is in the same org-roam file, when I push save the correctly formatted block turns into a UUID.
- If the block-reference is in a different org-roam file, when I change the original text, the reference-block turns into a UUID. 
To be clear, it's always the "copy" of the referenced block that changes into a UUID, never the original text. Going out and in of roam-block-mode turns currently working block-references into UUIDs, and does nothing to the other UUIDs.

Here is a copy of one of the test document text.

```
#+TITLE: Roam-block test 7

* Test 7

This is my test text from test 7. I hope this works.

((601bb653-1e5a-47b9-8826-b62f594ed372))
```

This UUID was a successfully displayed block-reference before I removed a period (.) from the original block and pushed save.

Any ideas what's happening here? Or suggestions on how I might figure it out? The code says that the package requires Emacs 26.1. I'm running 27.1. Is that okay does it mean that it must be 26.1?

Thank you,
Patrick.

**Comment by Kinneyzhang** [comment]
Sorry, I cannot reproduce the error. Try to call ```roam-block-overlay-buffer``` function manually in the test buffer to see what will happen. 

You could test in a new directory instead of in the org-roam directory to figure out whether exists conflicts with org-roam.

**Reply by PaddyPatPat** [reply]
Thanks for your reply.

I've just tried calling `roam-block-overlay-buffer` on my test document using `M-:`. It returned this debugger response:

```
Debugger entered--Lisp error: (void-variable roam-block-overlay-buffer)
  eval(roam-block-overlay-buffer t)
  eval-expression(roam-block-overlay-buffer nil nil 127)
  funcall-interactively(eval-expression roam-block-overlay-buffer nil nil 127)
  call-interactively(eval-expression nil nil)
  command-execute(eval-expression)
```

I tried moving my roam-block directory out of my org-roam directory, and the same error as above occurred when I ran `roam-block-overlay-buffer`.

**Reply by PaddyPatPat** [reply]
Org-roam started complaining about an sql error and kept failing to update it's database after an org-roam file save (although I'm still outside of the org-roam directory). 

I updated all the packages, but this didn't help either issue. Then I reinstalled emacs. Uninstalled and installed via D12frosted Emacs-plus 27 from HomeBrew.

The embedded blocks functionality is now working. The referencing also seems to work now. Seems I needed a reinstall to fix something...

**Reply by PaddyPatPat** [reply]
Hmmm ok. I still sometimes get the UUID's popping up but I can't figure out why yet. It might be something to do with where the limits of each block are. Dunno yet.

**Comment by Kinneyzhang** [comment]
> Debugger entered--Lisp error: (void-variable roam-block-overlay-buffer)

```roam-block-overlay-buffer``` is not a varibable, it's a function, so you should input ```(roam-block-overlay-buffer)``` after press M-: instead of ```roam-block-overlay-buffer```.

**Reply by PaddyPatPat** [reply]
Ahh thank you.

**Comment by Kinneyzhang** [comment]
> Hmmm ok. I still sometimes get the UUID's popping up but I can't figure out why yet. It might be something to do with where the limits of each block are. Dunno yet.

There do exist some bugs casually, but I have't got time to figure out and fix them.

**Reply by PaddyPatPat** [reply]
Of course. Thank you.
